verifier.go

```diff
@@ -35,9 +38,15 @@ func NewVerifier(issuer string, keySource KeySource) *Verifier {
 	}
 }
 
-type verifyCfg struct{}
+type VerifyOpt func(*Verifier)
 
-type VerifyOpt func(v *verifyCfg)
+// WithClock provides a custom clock that is used to determine the current time
+// when verifying tokens.
+func WithClock(clock func() time.Time) VerifyOpt {
+	return func(v *Verifier) {
+		v.clock = clock
+	}
+}
 
 func (v *Verifier) VerifyRaw(ctx context.Context, audience string, raw string, opts ...VerifyOpt) (*Claims, error) {
```

Comment:
nothing in this function appears to be calling the opts on the verifier, so might need to add something